e2e-tests/portal/cypress/support/utils/insertElement.js

```diff
@@ -15,12 +15,27 @@ const insertElement = (divId) => {
     textInput.setAttribute('id', 'intruder');
 
     // gets the buyer-name div which is part of the form
-    const div = win.document.getElementById(divId);
-
-    // adds element to div so is part of the form
-    div.appendChild(textInput);
-    // types text into the field
-    cy.keyboardInput(cy.get('[data-cy="intruder"]'), 'input text');
+    // wait for the element to be available in the DOM
+    cy.get('body').then(($body) => {
+      if ($body.find(`#${divId}`).length) {
+        cy.get(`#${divId}`)
+          .should('exist')
+          .then(($div) => {
+            const div = $div[0];
+            // adds element to div so is part of the form
+            div.appendChild(textInput);
+            // types text into the field
+            cy.keyboardInput(cy.get('[data-cy="intruder"]'), 'input text');
+          });
+      } else {
+        // Create the div if it doesn't exist
+        const newDiv = win.document.createElement('div');
+        newDiv.setAttribute('id', divId);
+        win.document.body.appendChild(newDiv);
+        newDiv.appendChild(textInput);
+        cy.keyboardInput(cy.get('[data-cy="intruder"]'), 'input text');
+      }
+    });
```

Comment:
Why has this been updated? I don't think this is correct. The previous logic simply appended a child and typed in some text.

Reply:
`appendChild` is returning null that's why I updated.